DOCS-3602 Dimensioning Private Locations Edits (#14432)
DataDog/documentation: content/en/synthetics/private_locations/dimensioning.md

To improve dimensioning, split your test assignments based on test types. For example, you can have some private locations run only API and multistep API tests while other private locations run only browser tests.

### Prerequisites

> You will need:

Author: @m-rousse @troisdiz can I get you both to review this when you have time? 🙏🏻

Reviewer: ```suggestion
To get started with dimensioning your private locations, you need the following:
```

1. Basic understanding of Docker and container orchestration

> 2. The private location configuration file mounted with your orchestrator of choice and accessible to your underlying private location containers

Reviewer: ```suggestion
2. The private location configuration file mounted with your orchestrator of choice and is accessible to your underlying private location containers.
```

> 3. If you are using browser tests with IP blocking (see [Advanced configuration][5]), `sudo` access may be required

Reviewer: ```suggestion
3. If you are using [browser tests with IP blocking][5], `sudo` access may be required.
```

| Test type                                     | CPU/Memory/Disk recommendation    |
| --------------------------------------------- | --------------------------------- |
| [API tests][1] and [Multistep API tests][2] | 100mCores/200MiB/100MiB per test run   |
| [Browser tests][3]                           | 800mCores/1GiB/500MiB per test run |

> For example, Datadog recommends ~ 8 core CPU `(150mCores + (150mCores*10 test runs))`, ~ 10GiB memory `(150MiB + (1GiB*10 test runs))`, and ~ 5GiB disk `(500MiB*10 test runs)` for a private location running only Browser tests with a maximum number of concurrent test runs of `10`.

Reviewer 2: ```suggestion
For example, Datadog recommends ~ 8 core CPU `(150mCores + (800mCores*10 test runs))`, ~ 10GiB memory `(150MiB + (1GiB*10 test runs))`, and ~ 5GiB disk `(500MiB*10 test runs)` for a private location running only Browser tests with a maximum number of concurrent test runs of `10`.
```

### Assign resources to your private location

Once you have determined the [total requirements for your private location](#define-your-total-hardware-requirements), decide how you want these resources to be distributed: by assigning all resources to a single worker or by distributing all resources across multiple workers.
To assign all resources to a single worker, run one container for a private location with a configuration file.
1. Set the [`concurrency` parameter][4] to `maximum number of test runs that can be executed in parallel on your private location`.
2. Assign your [total private location resource requirements](#define-your-total-hardware-requirements) to your unique container.
  
To distribute resources across multiple workers, run multiple containers for a private location with a configuration file.
 
 1. Set the [`concurrency` parameter][4] to `maximum number of test runs that can be executed on your private location / number of workers associated with your private location`.
 2. Assign `total private location resource requirements / number of workers` resources to every private location container.

> For example, Datadog recommends ~ 1.5 core CPU, ~ 10GiB memory, and ~ 100MiB disk for a private location running only Browser tests with a maximum number of concurrent test runs of `10`. To distribute these resources across two workers, set the [`concurrency` parameter][4] to 5 and allocate ~ 750mCores CPU, ~ 5GiB memory, and ~ 50MiB disk to each worker.

Reviewer 2: ```suggestion
For example, Datadog recommends ~ 8 cores CPU, ~ 10GiB memory, and ~ 5GiB disk for a private location running only Browser tests with a maximum number of concurrent test runs of `10`. To distribute these resources across two workers, set the [`concurrency` parameter][4] to 5 and allocate ~ 4 cores CPU, ~ 5GiB memory, and ~ 2.5GiB disk to each worker.
```

For example, ten tests are scheduled to run simultaneously on a private location that has two workers running. If Worker 1 is running two tests, Worker 1 can request three additional tests to run. If Worker 2 is not running any tests, Worker 2 can request the five following tests. The remaining two tests can be requested by whichever worker has finished running its test first (any worker that has available slots).

> ### Troubleshooting

Reviewer: How do you feel about adding this section into the Troubleshooting page (https://docs-staging.datadoghq.com/beth.glenfield/update-synthetics-docs/synthetics/troubleshooting/)?

Author: Let's do it! I wasn't sure where was appropriate 😄

#### Browser tests are taking too long to run

> Confirm you are not seeing [out of memory issues][6] with your private location deployments. If you have scaled your container instances following our [guidelines](#Define-your-total-hardware-requirements), please reach out to Datadog Support for assistance.

Reviewer: ```suggestion
Confirm you are not seeing [out of memory issues][6] with your private location deployments. If you have scaled your container instances following the [guidelines](#Define-your-total-hardware-requirements) above, reach out to [Datadog Support][7].
```

[3]: /synthetics/browser_tests/?tab=requestoptions
[4]: /synthetics/cicd_integrations
[5]: /synthetics/private_locations/configuration#advanced-configuration

> [6]: https://docs.docker.com/config/containers/resource_constraints/

Reviewer: ```suggestion
[6]: https://docs.docker.com/config/containers/resource_constraints/
[7]: /help/
```
